src/CborParseNode.cs

```diff
@@ -322,11 +333,22 @@ public IEnumerable<T> GetCollectionOfPrimitiveValues<T>()
         {
             var genericType = typeof(T);
             if(value is object?[] arrayValue1)
-                return arrayValue1.OfType<CborParseNode>().Select(x => GetItemValue<T>(genericType, x));
+                return GetValuesFromArray<T>(genericType, arrayValue1);
             else if(value is CborParseNode { value: object?[] arrayValue2 })
-                return arrayValue2.OfType<CborParseNode>().Select(x => GetItemValue<T>(genericType, x));
+                return GetValuesFromArray<T>(genericType, arrayValue2);
             else
                 return Enumerable.Empty<T>();
+        
+            IEnumerable<T> GetValuesFromArray<T>(Type genericType, object?[] array)
+            {
+                foreach (var item in array)
+                {
+                    if (item is CborParseNode node)
+                    {
+                        yield return GetItemValue<T>(genericType, node);
+                    }
+                }
+            }
```

Comment:
`T` matches outer function `T`
```suggestion
            static IEnumerable<TItem> GetValuesFromArray<TItem>(Type genericType, object?[] array)
            {
                foreach (var item in array)
                {
                    if (item is CborParseNode node)
                    {
                        yield return GetItemValue<TItem>(genericType, node);
                    }
                }
            }
```